otx/algorithms/common/tasks/training_base.py

```diff
@@ -101,6 +101,7 @@ def _run_task(self, stage_module, mode=None, dataset=None, **kwargs):
         # FIXME: Temporary remedy for CVS-88098
         export = kwargs.get("export", False)
         self._initialize(export=export)
+        stage_module = self._update_stage_module(stage_module)
```

Comment:
If it is called for train, does it use the method inside inference? And I saw the code 'inference.py', that it uses always 'DetectionInferrer' when calling infer_detector. Was it meant to be? Could you explain the workflow of this stage_module?

Reply:
_run_task function in the traininig_base.py is used both in inference and training task in detection, and both task do not modify this function. Inference and training task call _run_task function to link each task and mpa stage. Therefore this new _update_stage_module function change stage_module if the task is not normal incremental learning(e.g. semi-sl)

Reply:
That's right!! Thanks